lib/driver.js

```diff
@@ -399,6 +399,18 @@ class XCUITestDriver extends BaseDriver {
       }
     }
 
+    if (this.opts.permissions) {
+      if (this.isSimulator()) {
+        log.debug('Setting the requested permissions before WDA is started');
+        for (const [bundleId, permissionsMapping] of JSON.parse(this.opts.permissions)) {
```

Comment:
Below worked since we have no `setPermissions`. We have `setPermission`
```javascript
        for (const [bundleId, permissionsMapping] of _.toPairs(JSON.parse(this.opts.permissions))) {
          for (const key of Object.keys(permissionsMapping)) {
            // Error: Status value 'YES' is unknown. Only the following values are supported: ["unset","no","yes"]
            await this.opts.device.setPermission(bundleId, key, permissionsMapping[key].toLowerCase());
          }
        }
```

Reply:
ah, i understood. => https://github.com/appium/appium-ios-simulator/pull/190/files#diff-85f7ff9952e43947bd522da2a5a4c072R1145

I'll apply both change.

Reply:
Sorry, only below change was needed to run. (with the ios-simulator#190)
```js
for (const [bundleId, permissionsMapping] of JSON.parse(this.opts.permissions)) {
```
=>
```js
for (const [bundleId, permissionsMapping] of _.toPairs(JSON.parse(this.opts.permissions))) {
```

Reply:
Good catch. Fixed